File: 01/01-two-sum.js
// https://leetcode.com/problems/two-sum/description/
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    const alreadyChecked = {}
    for (let i = 0; i < nums.length; i++) {
        const curr = nums[i];
        const rest = target - curr;
        if (alreadyChecked.hasOwnProperty(rest)) {
            return [alreadyChecked[rest], i];
        }
        alreadyChecked[curr] = i;
    }
    return [-1, -1];
};

console.log(twoSum([2,7,11,15], 9))

File: 01-two-sum/01-two-sum.py
# https://leetcode.com/problems/two-sum/description/

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        alreadyChecked = {}
        for i in range(len(nums)):
            curr = nums[i]
            rest = target - curr
            if (rest in alreadyChecked):
                return [alreadyChecked[rest], i]
            alreadyChecked[curr] = i
        return [-1, -1]


print(Solution().twoSum([2, 7, 11, 15], 9))
